sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/encoders/ExpressionEncoderSuite.scala

```diff
@@ -597,6 +596,9 @@ class ExpressionEncoderSuite extends CodegenInterpretedPlanTest with AnalysisTes
         case (b1: Array[Int], b2: Array[Int]) => Arrays.equals(b1, b2)
         case (b1: Array[Array[_]], b2: Array[Array[_]]) =>
           Arrays.deepEquals(b1.asInstanceOf[Array[AnyRef]], b2.asInstanceOf[Array[AnyRef]])
+        case (b1: Array[Array[_]], b2: Array[_]) =>
+          // Since we expect array of array, use deepEquals
+          Arrays.deepEquals(b1.asInstanceOf[Array[AnyRef]], b2.asInstanceOf[Array[AnyRef]])
         case (b1: Array[_], b2: Array[_]) =>
           Arrays.equals(b1.asInstanceOf[Array[AnyRef]], b2.asInstanceOf[Array[AnyRef]])
```

Comment:
shall we simply change this line to use `deepEquals`?

Reply:
Sure!